sw/device/lib/dif/dif_spi_device.h

```diff
@@ -106,77 +78,6 @@ typedef struct dif_spi_device_config {
   uint16_t tx_fifo_len;
 } dif_spi_device_config_t;
```

Comment:
Could we call this `dif_spi_device_params_t` to stay consistent with the parameterized `alert_handler` DIFs?